### PR TITLE
Fix race when importing the release image stream

### PR DESCRIPTION
**Pull request by smarterclayton**
The image import controller races with the image pull secret controller
because namespace creation is occuring. Use the import API and retry a few
times.

Builds on #294 since that's required.

### CODE REVIEW

File: pkg/steps/release/create_release.go [hunk 1]
```diff
@@ -221,39 +221,43 @@ func (s *assembleReleaseStep) importFromReleaseImage(ctx context.Context, dry bo
...
-		var registry string
-		if len(is.Status.PublicDockerImageRepository) > 0 {
-			registry = is.Status.PublicDockerImageRepository
-		} else if len(is.Status.DockerImageRepository) > 0 {
-			registry = is.Status.DockerImageRepository
-		} else {
-			return false, fmt.Errorf("image stream %s has no accessible image registry value", is.Name)
-		}
-		pullSpec = fmt.Sprintf("%s:%s", registry, tag)
+		pullSpec = result.Status.Images[0].Image.DockerImageReference
```
Comment (stevekuznetsov) [inline pkg/steps/release/create_release.go]: Why doesn't this have to be public anymore

File: pkg/steps/release/release_images.go [hunk 2]
```diff
@@ -127,6 +127,16 @@ func findStatusTag(is *imageapi.ImageStream, tag string) (*coreapi.ObjectReferen
 	return nil, ""
 }
 
+func findTagReference(is *imageapi.ImageStream, name string) *imageapi.TagReference {
```
Comment (stevekuznetsov) [inline pkg/steps/release/release_images.go]: Where is this used?

File: pkg/steps/release/create_release.go (hunk 1 above)
Comment (smarterclayton) [reply pkg/steps/release/create_release.go]: Only used within this namespace - the pods will correctly pull from internal registry.  The bits that use it outside this job would continue to be public.

File: pkg/steps/release/release_images.go (hunk 2 above)
Comment (smarterclayton) [reply pkg/steps/release/release_images.go]: Ah, deleted code.

**Comment by smarterclayton** [comment]
https://openshift-gce-devel.appspot.com/build/origin-ci-test/logs/release-openshift-origin-installer-e2e-aws-serial-4.0/2640 is the symptom

**Comment by smarterclayton** [comment]
Fixed the unneeded code

**Comment by smarterclayton** [comment]
Tested by updating the latest tagged image - found one more issue here 
https://openshift-gce-devel.appspot.com/build/origin-ci-test/logs/release-openshift-origin-installer-e2e-aws-serial-4.0/2644

```
2019/03/10 01:42:13 Setting a soft TTL of 1h0m0s for the namespace
2019/03/10 01:42:13 Setting a hard TTL of 12h0m0s for the namespace
2019/03/10 01:42:13 Setting up pipeline imagestream for the test
2019/03/10 01:42:13 Created secret release-e2e-aws-cluster-profile
2019/03/10 01:42:13 Importing release image latest
2019/03/10 01:42:14 Ran for 1s
error: could not run steps: unable to find the 'cli' image in the provided release image: unable to create pod: pods "release-images-latest-cli" is forbidden: no providers available to validate pod request
```

This is because the namespace is still warming up - we generally don't run pods this fast normally (import takes a second or so), so will add an extra commit here to retry in this specific error mode.

All other jobs look green (and we got a passing release payload again).

**Comment by smarterclayton** [comment]
Fixed another: 
https://openshift-gce-devel.appspot.com/build/origin-ci-test/logs/release-openshift-origin-installer-e2e-aws-4.0/5470

```
unable to import latest release image: Operation cannot be fulfilled on imagestreamimports.image.openshift.io "release": the object has been modified; please apply your changes to the latest version and try again
```

That should be the last of them, upgrade jobs running now 
https://openshift-gce-devel.appspot.com/build/origin-ci-test/logs/release-openshift-origin-installer-e2e-aws-upgrade-4.0/6

**Comment by smarterclayton** [comment]
Note that `ci/ci-operator:latest` is set to the image from this PR right now so jobs can soak before Monday.

**Comment by smarterclayton** [comment]
Haven't noticed any problems today.

**Comment by smarterclayton** [comment]
I think this is ready for a final look - not seeing anything significant.

**Pull request merged**